sktime/tests/test_all_estimators.py

```diff
@@ -878,7 +877,8 @@ def test_constructor(self, estimator_class):
         assert isinstance(estimator, estimator_class)
 
         # Ensure that each parameter is set in init
-        init_params = _get_args(type(estimator).__init__)
+        # init_params = _get_args(type(estimator).__init__)
+        init_params = vars(estimator)
```

Comment:
Whoops, this was a small workaround I had put into place to deal with the fact that I had constructed a different __init__ (via define).  Will fix this before merging

Reply:
ok, I just scrolled down and do see that the little `define` thing causes errors...

we can't change the core interface contract because of that.
`vars`, for instance, does not return all init args, so this change breaks the test!

Reply:
I totally agree, it was a spandrel I left in by mistake.  Currently working to see if I can still make the `attrs` work despite this though! :)

Reply:
well, I've also tried and failed...
https://github.com/alan-turing-institute/sktime/pull/2735